### PR TITLE
kmeshctl dump failed issue resolve

### LINKED ISSUE #1016: `kmeshctl dump` failed
**Task description**:

`kmeshctl dump` failed to execute becasuse rename of Kmesh mode from `ads` & `workload` to `kernel-native` & `dual-engine`

```bash
root@dev:~/kmesh# kmeshctl dump kmesh-ctq7d dual-engine 
404 page not found
```

**Solution**:

The mode in the url should also be updated:

https://github.com/kmesh-net/kmesh/blob/main/pkg/status/status_server.go#L53-L57

friendly ping @codesmith25103 if you're interested :)

***If no one claims this for a while, I will do it myself, because this subcommand is very commonly used in the dev process***

**Who can join or take the task**:

The good first issue is intended for `first-time contributors` to get started on his/her contributor journey.

After a contributor has successfully completed 1-2 good first issue's,
they should be ready to move on to `help wanted` items, saving the remaining `good first issue` for other new contributors.

**How to join or take the task**:

Just reply on the issue with the message `/assign` in a separate line.

Then, the issue will be assigned to you.

**How to ask for help**:

If you need help or have questions, please feel free to ask on this issue.
The issue author or other members of the community will guide you through the contribution process.

### PR DESCRIPTION
**Pull request by codesmith25103**
**What type of PR is this?**
Kmesh mode has been successfully renamed from ads & workload to kernel-native & dual-engine

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #1016 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/status/status_server.go [hunk 1]
```diff
@@ -50,11 +50,11 @@ const (
...
 	patternVersion            = "/version"
-	patternBpfAdsMaps         = "/debug/config_dump/bpf/ads"
-	patternBpfWorkloadMaps    = "/debug/config_dump/bpf/workload"
+	patternBpfAdsMaps         = "/debug/config_dump/bpf/kernel-native"
+	patternBpfWorkloadMaps    = "/debug/config_dump/bpf/dual-engine"
 	configDumpPrefix          = "/debug/config_dump"
-	patternConfigDumpAds      = configDumpPrefix + "/ads"
-	patternConfigDumpWorkload = configDumpPrefix + "/workload"
+	patternConfigDumpAds      = configDumpPrefix + "/kernel-native"
+	patternConfigDumpWorkload = configDumpPrefix + "/dual-engine"
```
Comment (YaoZengzeng) [inline pkg/status/status_server.go:57]: Good fix, it would be better to add UT for this subcommand, then similar issue would not occur.

Comment (LiZhenCheng9527) [inline pkg/status/status_server.go:57]: The error should not be exposed directly if kernel-native mode is used in duel-engine mode. 
The user should be prompted that cannot use the kernel-native command dump map in `duel-engine` mode.
So I recommend optimising the error message.

Comment (codesmith25103) [reply pkg/status/status_server.go:57]: Yeah sure. I will add UT for this command.

Comment (codesmith25103) [reply pkg/status/status_server.go:57]: Ok, I will optimise the error message

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**